march_joint_inertia_controller/src/inertia_estimator.cpp

```diff
@@ -0,0 +1,290 @@
+// Copyright 2020 Project March.
+#include <hardware_interface/joint_command_interface.h>
+#include <hardware_interface/joint_state_interface.h>
+#include "march_joint_inertia_controller/inertia_estimator.h"
+#include "math.h"
+#include "ros/ros.h"
+#include "std_msgs/Float64.h"
+
+void absolute(std::vector<double> a, std::vector<double>& b)
+{
+  for (size_t i = 0; i < a.size(); ++i)
+  {
+    b[i] = abs(a[i]);
+  }
+}
+
+double mean(std::vector<double> a)
+{
+  double sum = 0.0;
+  for (size_t i = 0; i < a.size(); ++i)
+  {
+    sum += a[i];
+  }
+  if (sum == 0.0)
+  {
+    return 0.0;
+  }
+  return sum / a.size();
+}
+
+InertiaEstimator::InertiaEstimator(double lambda, size_t acc_size)
+{
+  lambda_ = lambda;
+  acc_size_ = acc_size;
+  joint_inertia_ = 0.0;
+
+  // Size the buffers
+  velocity_array_.resize(vel_size_);
+
+  acceleration_array_.resize(acc_size_);
+
+  filtered_acceleration_array_.resize(acc_size_);
+
+  joint_torque_.resize(torque_size_);
+
+  filtered_joint_torque_.resize(fil_tor_size_);
```

Comment:
To be completely sure they are all initialized to 0
```suggestion
  velocity_array_.resize(vel_size_, 0.0);

  acceleration_array_.resize(acc_size_, 0.0);

  filtered_acceleration_array_.resize(acc_size_, 0.0);

  joint_torque_.resize(torque_size_, 0.0);

  filtered_joint_torque_.resize(fil_tor_size_, 0.0);
```